Glossary View OMAS - client added

lcpopa opened the pull request:
Signed-off-by: Liviu Popa <liviu-constantin.popa@ing.com>

mandy-chessell commented on open-metadata-implementation/access-services/glossary-view/glossary-view-api/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/exception/GlossaryViewOmasException.java:
```diff
@@ -7,14 +7,14 @@
  * Stores checked OMRS exceptions info until it is sent to the client. Made to be used only by {@code OMRSClient}
  */
-public class OMRSExceptionWrapper extends OMRSCheckedExceptionBase {
+public class GlossaryViewOmasException extends OMRSCheckedExceptionBase {
```
OMAS Exceptions should extend the common services exceptions or OCF

lcpopa replied:
Changed extended class

mandy-chessell commented on open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java:
```diff
@@ -0,0 +1,599 @@
+    private static final String GET_SUBTYPES = PATH_ROOT + "/terms/{2}/subtypes" + PARAM_DELIMITER + PAGINATION;
+    private static final String GET_TYPES = PATH_ROOT + "/terms/{2}/types" + PARAM_DELIMITER + PAGINATION;
+
+    /**
```
Add constructor with no userId and password

lcpopa replied:
Added

mandy-chessell commented on open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java:
```diff
@@ -0,0 +1,599 @@
+     * Extract all glossary definitions
+     *
+     * @param userId calling user
+     * @param serverName instance to call
```
Not all parameters are included in Javadoc

lcpopa replied:
Revised all methods in the client and added the javadoc for the missing params

mandy-chessell commented on open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java:
```diff
@@ -0,0 +1,599 @@
+     * Extract a term's home glossary
+     *
+     * @param userId calling user
+     * @param serverName instance to call
```
ServerName should be in constructor - this applied to other calls.

lcpopa replied:
Removed serverName from method signatures and switched usage to field

mandy-chessell commented on open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java:
```diff
@@ -0,0 +1,599 @@
+     * @throws InvalidParameterException if parameter validation fails
+     * @throws GlossaryViewOmasException if a problem occurs on the omas backend
+     */
+    public GlossaryViewEntityDetailResponse getTermsOfGlossary(String serverName, String userId, String glossaryGUID,
```
The result of the client calls should be a standard Java interface with the beans returned and the exceptions called out.  The response object contains the exception information which should be returned as exceptions

lcpopa replied:
Changed return types from wrapper to specific pojos

mandy-chessell commented on open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/OmasClient.java:
```diff
@@ -0,0 +1,79 @@
+        GlossaryViewEntityDetailResponse response = callGetRESTCall(methodName, GlossaryViewEntityDetailResponse.class,
+                serverPlatformURLRoot + path, serverName, userId, entityGUID, from, size);
+
+        exceptionHandler.detectAndThrowOMRSException(methodName, response);
```
Not all exceptions are from OMRS - infact there should be NO OMRS exceptions returned to an OMAS client.  The exceptions returned should be common services, OMAS or OCF exceptions.

lcpopa replied:
Changed method name to reflect the change of exception type

lcpopa commented:
Corrected code based on previous observations, but I have also introduced some small modifications:
- setting the type returned to 'ExternalGlossaryLink', since it was not being returned by targeted calls
- extracted fields to classes for better readability and access to logic
- pom updates
- renamed some classes for consistency
- simplified exception handling
- java doc update
- code improvements as a result of sonar scan
